tools/build_config/Jenkinsfile

```diff
@@ -329,15 +329,32 @@ pipeline {
       steps {
         script {
           if (isUnix()) {
+            if (env.BUILD_DOCS?.toBoolean()) {
```

Comment:
why does analyse need to be aware of whether the docs are being built? We're not performing any static analysis on the documents

Reply:
This was more about not wanting to have to define a new configure stage. Because configure tries to find `sphinx-build` here, if it is not installed it will set that to be not found. This is a problem when it comes to build the docs.

Reply:
the complexity and unnecessary configuration this has introduced suggests extracting a separate configure stage with knowledge of whether the doc stage is required would be clearer